--- barebox-2017.08.0/arch/arm/dts/socfpga_cyclone5_de0_nano_soc.dts.orig	2017-08-25 08:46:16.655463004 +0200
+++ barebox-2017.08.0/arch/arm/dts/socfpga_cyclone5_de0_nano_soc.dts	2017-08-25 08:45:28.446405834 +0200
@@ -20,7 +20,7 @@
 
 / {
 	model = "Terasic DE0-Nano-SoC/Atlas-SoC Kit";
-	compatible = "terasic,de0-nano-soc","altr,socfpga-cyclone5", "altr,socfpga";
+	compatible = "terasic,de0-atlas","altr,socfpga-cyclone5", "altr,socfpga";
 
 	chosen {
 		stdout-path = &uart0;
